SX1276/SX1276_LoRaRadio.cpp

```diff
@@ -718,25 +718,18 @@ uint32_t SX1276_LoRaRadio::time_on_air(radio_modems_t modem, uint8_t pkt_len)
             }
 
             // Symbol rate : time for one symbol (secs)
-            float rs = bw / (1 << _rf_settings.lora.datarate);
-            float ts = 1 / rs;
-            // time of preamble
-            float tPreamble = (_rf_settings.lora.preamble_len + 4.25f) * ts;
+            uint32_t tPreamble = (uint32_t) (((uint64_t) (((_rf_settings.lora.preamble_len * 1000) + 4250) * (1 << _rf_settings.lora.datarate))) / bw); // unit = ms
+
             // Symbol length of payload and time
-            float tmp = ceil((8 * pkt_len - 4 * _rf_settings.lora.datarate + 28
-                            + 16 * _rf_settings.lora.crc_on
-                            - (_rf_settings.lora.fix_len ? 20 : 0))
-                            / (float) (4
-                                    * (_rf_settings.lora.datarate
-                                            - ((_rf_settings.lora.low_datarate_optimize > 0)
-                                                    ? 2 : 0))))
-                            * (_rf_settings.lora.coderate + 4);
-            float nPayload = 8 + ((tmp > 0) ? tmp : 0);
-            float tPayload = nPayload * ts;
+            int64_t tmp1 = 1000 * 8 * pkt_len - 4 * _rf_settings.lora.datarate + 28 + 16 * _rf_settings.lora.crc_on - (_rf_settings.lora.fix_len ? 20 : 0);
```

Comment:
Minor suggestion: you could use pkt_len_ms = pkt_len * 1000 in order to keep a similar syntax as before